cmd/get.go

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2023 Genome Research Ltd.
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package cmd
+
+import (
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/wtsi-hgi/ibackup/baton"
+	"github.com/wtsi-hgi/ibackup/put"
+)
+
+var overwrite bool
+
+// putCmd represents the put command.
+var getCmd = &cobra.Command{
+	Use:   "get",
+	Short: "Safely copy files from iRODS to a local filesystem",
+	Long: `Safely copy files from iRODS to a local filesystem.
+
+Use the -f arg (defaults to STDIN) to provide a tab-delimited file with these 2
+columns:
+/absolute/path/of/local/file /remote/irods/path/of/file
+
+Because local and remote paths could contain tabs and newlines in their names,
+you can base64 encode them and use the --base64 option.
+
+get will then efficiently copy all column 1 paths in iRODS to column 2
+locations, using a single connection, sequentially. (This makes it orders of
+magnitude faster than running 'iput' for each of many small files.)
+
+get will always confirm the checksum stored on the server side in flight, but
+will not confirm the data stored on disk.
+
+It will not overwrite existing local files unless the -o/--overwrite flag is
+provided.
+
```

Comment:
Document something about the behaviour of skipping downloads if the file exists with same mtime, even with -o. Represent this as a good thing, eg. "You can repeat a get on the same input and it will skip files it already got".